### ISSUE TITLE
CMakeUserPresets regression

### ISSUE THREAD

**Issue by christophe-calmejane** [microsoft/vscode-cmake-tools#3956]
### Brief Issue Summary

CMakeUserPresets is no longer working when extending an existing configure preset.
Let's take this very short example:

`CMakePresets.json` file:
```json
{
	"version": 6,
	"configurePresets": [
		{
			"name": "project",
			"binaryDir": "${sourceDir}/_build_${presetName}"
		}
	]
}
```

and this `CMakeUserPresets.json` file:
```json
{
	"version": 6,
	"configurePresets": [
		{
			"name": "user project",
			"inherits": [
				"project"
			]
		}
	]
}
```

When I run `cmake --list-presets` I correctly get:
```
Available configure presets:

  "user project"
  "project"
```

Since a recent update of vccode-cmake-tools, I don't get my `user project` anymore.

As a reminder, this is taken from cmake documentation:
`If CMakePresets.json and CMakeUserPresets.json are both present, CMakeUserPresets.json implicitly includes CMakePresets.json, even with no include field, in all versions of the format.`

Although I did try to add an include of CMakePresets in my CMakeUserPresets file, it doesn't help.

### Additional Information

I'm on v1.19.35

**Comment by gcampbell-msft** [comment]
Could you test this vsix? 
[cmake-tools.zip](https://github.com/user-attachments/files/16494670/cmake-tools.zip)

This should (I believe), fix the regression.

**Comment by christophe-calmejane** [comment]
It doesn't look like a vsix file, how should I install it?

**Comment by gcampbell-msft** [comment]
@christophe-calmejane Ah, my apologies, some people I've already given the instructions and some I haven't. 

Steps:
1. Download the file.
2. Change the extension from .zip to .vsix (Github doesn't allow dropping of .vsix)
3. Install manually in VSCode extensions pane:
![image](https://github.com/user-attachments/assets/2a4df698-b1e1-40b7-b4b4-bd53f4339738)

**Comment by christophe-calmejane** [comment]
Oh yes sorry I didn't know that. (But I just dropped the vsix file to the extension panel and it correctly replaced my version with yours).
I can confirm it both fixes this issue, and that #3522 is also working fine.
